### PR TITLE
analog: add Hour, Day uplink commands

### PR DESCRIPTION
**Pull request by DeadByElpy**

### CODE REVIEW

File: src/analog/header.ts [hunk 1]
```diff
@@ -51,7 +51,16 @@ export const fromBytes = ( data: Uint8Array ) => {
...
-export const toBytes = ( commandId: number, commandSize: number ): Uint8Array => {
+export const toBytes = ( commandId: number, commandSize: number, legacyHeader = false ): Uint8Array => {
+    if ( legacyHeader ) {
+        // legacy commands contain ID more than extra command mask, i.e. 0x1f so ignore all rules and return ID + size as two bytes as is
+        return new Uint8Array([
+            commandId,
+            commandSize
+        ]);
+    }
+
```
Comment (DarkPark) [inline src/analog/header.ts:62]:
```suggestion
```

File: src/analog/Command.ts [hunk 2]
```diff
@@ -91,9 +91,9 @@ abstract class Command {
      * @param commandData optional command binary data
      * @returns merged data
      */
-    static toBytes ( id: number, commandData?: Uint8Array ): Uint8Array {
+    static toBytes ( id: number, commandData?: Uint8Array, legacyHeader?: boolean ): Uint8Array {
```
Comment (DarkPark) [inline src/analog/Command.ts:94]:
```suggestion
    static toBytes ( id: number, commandData?: Uint8Array, isLegacyHeader?: boolean ): Uint8Array {
```
?

File: src/analog/CommandBinaryBuffer.ts [hunk 3]
```diff
@@ -1291,6 +1304,23 @@ class CommandBinaryBuffer extends BinaryBuffer {
         }));
     }
 
+    private getLegacyHourDiff (): ILegacyCounter {
+        const bytes = [this.getUint8(), this.getUint8()];
```
Comment (DarkPark) [inline src/analog/CommandBinaryBuffer.ts:1308]:
```suggestion
        const [someName, anotherName] = [this.getUint8(), this.getUint8()];
```
?

File: src/analog/CommandBinaryBuffer.ts [hunk 4]
```diff
@@ -1477,6 +1507,42 @@ class CommandBinaryBuffer extends BinaryBuffer {
         this.setUint8(CommandBinaryBuffer.setMagneticInfluenceBit(byte, counter.isMagneticInfluence));
         this.setLegacyCounterValue(counter.value);
     }
+
+    getLegacyHourCounterWithDiff (): ILegacyHourCounterWithDiff {
+        const date = this.getDate();
+        const byte = this.getUint8();
```
Comment (DarkPark) [inline src/analog/CommandBinaryBuffer.ts:1513]: you don't have to create a var if you don't need it

```suggestion
        const isMagneticInfluence = CommandBinaryBuffer.getMagneticInfluenceBit(this.getUint8());
```

Comment (DeadByElpy) [reply src/analog/CommandBinaryBuffer.ts:1513]: it's needed for 2 usages: extract start hour from this byte + extract `isMagneticInfluence` bit

File: src/analog/CommandBinaryBuffer.ts (hunk 3 above)
Comment (DeadByElpy) [reply src/analog/CommandBinaryBuffer.ts]: names added but don't know what they were for, no good variants for them

File: src/analog/Command.ts (hunk 2 above)
Comment (DeadByElpy) [reply src/analog/Command.ts]: fixed

File: src/analog/header.ts (hunk 1 above)
Comment (DeadByElpy) [reply src/analog/header.ts:62]: fixed

**Pull request merged**